src/components/progress-bar/progress-bar.scss

```diff
@@ -0,0 +1,265 @@
+      left: -145.166611%;
+    }
+    .primary.bar::after {
+      animation: md-progress-bar-primary-indeterminate-scale $md-progress-bar-full-animation-duration infinite linear;
+    }
+    .secondary {
+      animation: md-progress-bar-secondary-indeterminate-translate $md-progress-bar-full-animation-duration infinite linear;
+      left: -54.888891%;
+      visibility: visible;
+    }
+    .secondary.bar::after {
+      animation: md-progress-bar-secondary-indeterminate-scale $md-progress-bar-full-animation-duration infinite linear;
+    }
+  }
+
+  &[mode="buffer"] {
+    .background {
+      animation: md-progress-bar-background-scroll $md-progress-bar-piece-animation-duration infinite linear;
+      visibility: visible;
+    }
+  }
+}
+
+
+// Reverse the apparent directionality of progress vars for rtl.
+:host-context([dir="rtl"]) {
+  transform: rotateY(180deg);
+}
+
+
+/** The values used for animations in md-progress-bar, both timing and transformation, can be considered magic values.
+    They are sourced from the Material Design example spec and duplicate the values of the original designers
+    definitions. */
+
+// Progress Bar Timing functions:
+// $md-progress-bar-primary-indeterminate-translate-step-1 has no timing function.
+$md-progress-bar-primary-indeterminate-translate-step-2: cubic-bezier(.5, 0, .701732, .495819) !default;
+$md-progress-bar-primary-indeterminate-translate-step-3: cubic-bezier(.302435, .381352, .55, .956352) !default;
+// $md-progress-bar-primary-indeterminate-translate-step-4 has no timing function.
+
+// $md-progress-bar-primary-indeterminate-scale-step-1 has no timing function
+$md-progress-bar-primary-indeterminate-scale-step-2: cubic-bezier(.334731, .124820, .785844, 1) !default;
+$md-progress-bar-primary-indeterminate-scale-step-3: cubic-bezier(.06, .11, .6, 1) !default;
+// $md-progress-bar-primary-indeterminate-scale-step-4 has no timing function
+
+$md-progress-bar-secondary-indeterminate-translate-step-1: cubic-bezier(.15, 0, .515058, .409685) !default;
+$md-progress-bar-secondary-indeterminate-translate-step-2: cubic-bezier(.310330, .284058, .8, .733712) !default;
+$md-progress-bar-secondary-indeterminate-translate-step-3: cubic-bezier(.4, .627035, .6, .902026) !default;
+// $md-progress-bar-secondary-indeterminate-translate-step-4 has no timing function
+
+$md-progress-bar-secondary-indeterminate-scale-step-1: cubic-bezier(.15, 0, .515058, .409685) !default;
+$md-progress-bar-secondary-indeterminate-scale-step-2: cubic-bezier(.310330, .284058, .8, .733712) !default;
+$md-progress-bar-secondary-indeterminate-scale-step-3: cubic-bezier(.4, .627035, .6, .902026) !default;
+// $md-progress-bar-secondary-indeterminate-scale-step-4 has no timing function
+
+/** Animations for indeterminate and query mode. */
+// Primary indicator.
+@keyframes md-progress-bar-primary-indeterminate-translate {
```

Comment:
Sure can :smiley: 

The indeterminate state is essentially made up of two progress bars, one **primary** (the one that is shown in both the determinate and indeterminate states) and one **auxiliary**, which essentially mirrors the primary progress bar in appearance but is only shown to assist with the indeterminate animations.

Given the [mocks](https://material-design.storage.googleapis.com/publish/material_v_4/material_ext_publish/0B0NGgBg38lWWYmNmallST001a1k/components-progressactivity-typesofindicators-061101_Linear_Sheet_xhdpi_003.webm) for the indeterminate APIs, you essentially have 4 animations being performed in tandem:

Keyframe Block | Description
--------------------- | ---------------
`primary-indeterminate-translate` | Translation of the primary progressbar across the screen
`primary-indeterminate-scale` | Scaling of the primary progressbar as it's being translated across the screen
`auxiliary-indeterminate-translate` | Translation of the auxiliary progressbar across the screen
`auxiliary-indeterminate-scale` | Scaling of the auxiliary progressbar as it's being translated across the screen

Because two different transform animations need to be applied at once, the translation is applied to the outer element and the scaling is applied to the inner element, which provides the illusion necessary to make the animation work.